lib/Common/Core/SysInfo.h

```diff
@@ -62,7 +63,11 @@ class AutoSystemInfo : public SYSTEM_INFO
 #ifdef _WIN32
     static HMODULE GetCRTHandle();
 #endif
+#if defined(__APPLE__) && defined(_M_ARM64)
```

Comment:
"Apple" + "ARM64" can mean both iPhone and Arm-based Mac, I wonder if it is necessary to distinguish the two (I don't think we are trying to support iPhone).

Reply:
This is a good point, some of my #ifdefs may not be the most consistent, I don't think the codebase as a whole will run on an iPhone at the moment but I haven't tried.

Reply:
AFAIK M1/M2 are essentially just souped up A-series processors and ARM macOS can even run iOS apps natively so there's probably no need to distinguish.

Reply:
I suspect it won't run for one reason or the other, but this is probably low priority at the moment